src/libcore/tests/test_xml.py

```diff
@@ -298,3 +298,166 @@ def test20_upgrade_tree(variant_scalar_rgb):
                                <float name="intIOR" value="1.33"/>
                            </bsdf>
                        </scene>""")
+
+@fresolver_append_path
+def test21_xml_save_plugin(variant_scalar_rgb):
+    from mitsuba.core import xml
+    from mitsuba.python.dict_to_xml import WriteXML
+    import os
+    abs_path = os.path.abspath('resources/data/scenes/dict/dict.xml')
+    writer = WriteXML(abs_path)
+    scene_dict = {
+            "type": "sphere",
+            "center" : [0, 0, -10],
+            "radius" : 10.0,
+        }
+    writer.configure(scene_dict)
+    s1 = xml.load_dict({
+        'type': 'scene',
+        'sphere':{
+            "type": "sphere",
+            "center" : [0, 0, -10],
+            "radius" : 10.0,
+        }
+        })
+    s2 = xml.load_file(abs_path)
+    assert str(s1) == str(s2)
+
+@fresolver_append_path
+def test22_xml_missing_type(variant_scalar_rgb):
+    from mitsuba.python.dict_to_xml import WriteXML
+    import os
+    abs_path = os.path.abspath('resources/data/scenes/dict/dict.xml')
+    writer = WriteXML(abs_path)
+    scene_dict = {
+        'my_bsdf':{
+            'type':'diffuse'
+        }
+    }
+    with pytest.raises(ValueError) as e:
+        writer.configure(scene_dict)
+    e.match("Missing key: 'type'!")
+
+@fresolver_append_path
+def test23_xml_references(variant_scalar_rgb):
+    from mitsuba.core import xml
+    from mitsuba.python.dict_to_xml import WriteXML
+    import os
+    abs_path = os.path.abspath('resources/data/scenes/dict/dict.xml')
+    writer = WriteXML(abs_path)
+    scene_dict = {
+        'type': 'scene',
+        'bsdf1': {
+            'type': 'diffuse',
+            'reflectance': {
+                'type': 'rgb',
+                'value': [1.0, 0.0, 0.0]
+            }
+        },
+        'texture1': {
+            'type': 'checkerboard'
+        },
+        'bsdf4': {
+            'type': 'dielectric',
+            'specular_reflectance': {
+                'type': 'ref',
+                'id': 'texture1'
+            },
+            'id': 'bsdf2'
+        },
+        'shape1': {
+            'type': 'sphere',
+            'bsdf': {
+                'type': 'ref',
+                'id': 'bsdf1'
+            }
+        },
+        'shape2': {
+            'type': 'cylinder',
+            'bsdf': {
+                'type': 'ref',
+                'id': 'bsdf2'
+            }
+        }
+    }
+
+    s1 = xml.load_dict(scene_dict)
+    writer.configure(scene_dict)
+    s2 = xml.load_file(abs_path)
+
+    assert str(s1.shapes()[0].bsdf()) == str(s2.shapes()[0].bsdf())
+
+@fresolver_append_path
+def test_24_xml_point(variant_scalar_rgb):
+    from mitsuba.core import xml, Point3f
+    from mitsuba.python.dict_to_xml import WriteXML
+    import os
```

Comment:
Not sure the `os` is needed in those tests. Maybe you should use Mitsuba's filesystem instead?